Framework/PythonInterface/plugins/functions/PrimStretchedExpFT.py

```diff
@@ -0,0 +1,181 @@
+
+
+class PrimStretchedExpFT(IFunction1D):
+    # Class variables
+    _planck_constant = scipy.constants.Planck/scipy.constants.e*1E15  # meV*psec
+
+    # pylint: disable=super-on-old-class
+    def __init__(self):
+        """declare some constants"""
+        super(PrimStretchedExpFT, self).__init__()
+        self._parmList = list()
+
+    def category(self):
+        return 'QuasiElastic'
+
+    def init(self):
+        """Declare parameters that participate in the fitting"""
+        # Active fitting parameters
+        self.declareParameter('Height', 0.1, 'Intensity at the origin')
+        self.declareParameter('Tau', 100.0, 'Relaxation time')
+        self.declareParameter('Beta', 1.0, 'Stretching exponent')
+        self.declareParameter('Centre', 0.0, 'Centre of the peak')
+        # Keep order in which parameters are declared. Should be a class
+        # variable but we initialize it just below parameter declaration
+        # to make sure we follow the order.
+        self._parmList = ['Height', 'Tau', 'Beta', 'Centre']
+
+    def validateParams(self):
+        """Check parameters are positive"""
+        height = self.getParameterValue('Height')
+        tau = self.getParameterValue('Tau')
+        beta = self.getParameterValue('Beta')
+        Centre = self.getParameterValue('Centre')
+
+        for _, value in {'Height': height, 'Tau': tau, 'Beta': beta}.items():
+            if value <= 0:
+                return None
+        return {'Height': height, 'Tau': tau, 'Beta': beta, 'Centre': Centre}
+
+    def function1D(self, xvals, **optparms):
+        """ Fourier transform of the symmetrized stretched exponential integrated
+        within each energy bin.
+
+        The Symmetrized Stretched Exponential:
+                height * exp( - |t/tau|**beta )
+
+        The Fourier Transform:
+            F(E) \int_{-infty}^{infty} (dt/h) e^{-i2\pi Et/h} f(t)
+            F(E) is normalized:
+                \int_{-infty}^{infty} dE F(E) = 1
+
+        Let P(E) be the primitive of F(E) from minus infinity to E, then for element i of
+        xvals we compute:
+            1. bin_boundaries[i] = (xvals[i]+xvals[i+1])/2
+            3. P(bin_boundaries[i+1])- P(bin_boundaries[i])
+        :param xvals: list of values where to evaluate the function
+        :param optparms: alternate list of function parameters
+        :return: P(bin_boundaries[i+1])- P(bin_boundaries[i]), the difference of the primitive
+        """
+        p = self.validateParams()
+        if not p:
+            # return zeros if parameters not valid
+            return np.zeros(len(xvals), dtype=float)
+        # override with optparms (used for the numerical derivative)
+        if optparms:
+            for name in optparms.keys():
+                p[name] = optparms[name]
+
+        ne = len(xvals)
+        # energy spacing. Assumed xvals is a single-segment grid
+        # of increasing energy values
+        refine_factor = 16.0
```

Comment:
refine_factor effectively partitions each energy bin, so that we can calculate the integrand over a series of points within each energy bin. These values will later be summed up to calculate the integral. The bigger refine_factor, the more accurate the integral and the slower the evaluation of the whole fit function. The current value is a good compromise.